vllm/model_executor/layers/fused_moe/layer.py

```diff
@@ -642,9 +648,6 @@ def _load_w13(self,
             expert_data = expert_data.narrow(shard_dim, shard_size, shard_size)
         expert_data.copy_(loaded_weight)
 
-        if is_hpu and isinstance(self.quant_method, UnquantizedFusedMoEMethod):
-            self.moe_op.w13_list[expert_id].set_weight(orig_exp_data)
```

Comment:
Since llama4 is loading full instead of loading expert 1 by 1, set weight for VllmMixureOfExpert weight in this function is a little tricky.
Because of that, we moved the function to process_weights_after_loading.